core/arch/arm/include/kernel/thread.h

```diff
@@ -617,6 +617,24 @@ void thread_rpc_free_payload(uint64_t cookie, struct mobj *mobj);
  */
 uint32_t thread_rpc_cmd(uint32_t cmd, size_t num_params,
 		struct optee_msg_param *params);
+/**
+ * Allocates data for payload buffers.
+ * Buffer is exported using dma-buf API
+ *
+ * @size:	size in bytes of payload buffer
+ * @cookie:	returned cookie used when freeing the buffer
+ *
+ * @returns	mobj that describes allocated buffer or NULL on error
+ */
+struct mobj *thread_rpc_alloc_dma(size_t size, uint64_t *cookie);
```

Comment:
`alloc_dma` is misleading to dma capabilities.
Maybe rename to `thread_rpc_alloc_dmabuf_payload()`.

Is it a generic service or should the REE client (benchmark stack) initiate the specific allocation and provide the buffers, with any format, to the TEE?

Reply:
@etienne-lms 
> should the REE client (benchmark stack) initiate the specific allocation and provide the buffers, with any format, to the TEE?

Moving of timestamp buffer allocation procedure from benchmark CA to benchmark PTA was done because of two reasons:
1. To obtain manual control over ts buffer mapping in OP-TEE core VA space (using mobj API)
In case if timestamp buffer is allocated by benchmark CA and new register
user memory API, there is no possibility to keep this mapping permanent (because of https://github.com/OP-TEE/optee_os/blob/master/core/arch/arm/tee/entry_std.c#L199) among different TEEC_InvokeCommand invocations.
2. To be able to share timestamp buffer among different CAs (at least two: benchmark CA, which is in charge of consuming timestamps; and particular CA, which produces timestamps), I decided to use existing `tee_shm_alloc` explicitly specifying that `dma_buf_mgr` pool manager should be used (as in this case buffers are exported and can be mmaped)

> Is it a generic service 

Frankly, I do not have on my mind any other cases where it could be used, except of this